lib/tapioca/runtime/trackers/constant_definition.rb

```diff
@@ -17,41 +17,60 @@ class ConstantLocation < T::Struct
           const :path, String
         end
 
-        @class_files = {}
+        @class_files = {}.compare_by_identity
 
         # Immediately activated upon load. Observes class/module definition.
-        TracePoint.trace(:class) do |tp|
-          unless tp.self.singleton_class?
-            key = name_of(tp.self)
-            file = tp.path
-            lineno = tp.lineno
+        TracePoint.trace(:class, :c_return) do |tp|
```

Comment:
This is entirely a suggestion, feel free to disregard - I personally find the single TracePoint with the `case` statement hard to parse. Some of the lines are pretty deeply nested, and the two cases don't really share any common functionality, aside from setting the key-value-pairs in `@class_files`.

What would you think about calling `TracePoint` twice, separating out the behavior for `class` and `c_return`, and then extracting `@class_files[key] ||= Set.new ...` into a shared method? I personally would find that much easier to read, but I don't think it's a huge deal either way.

(I'm also curious if there's any performance benefit to calling T`TracePoint` once?)

```suggestion
TracePoint.trace(:class) do |tp|
  next if tp.self.singleton_class?

  if tp.path == "(eval)"
    caller_location = T.must(caller_locations)
      .drop_while { |loc| loc.path == "(eval)" }
      .first

    next unless caller_location

    loc = ConstantLocation.new(
      path: caller_location.absolute_path || "",
      lineno: caller_location.lineno
    )
  else
    loc = build_constant_location(tp, caller_locations)
  end

  add_to_class_files(tp.self, loc)
end

TracePoint.trace(:c_return) do |tp|
  next unless tp.method_id == :new
  next unless Module === tp.return_value

  key = tp.return_value
  loc = build_constant_location(tp, caller_locations)

  add_to_class_files(key, loc)
end
```